docs/docs/08.1-more-about-refs.md

```diff
@@ -86,75 +87,40 @@ It's as simple as:
   You can access the component's DOM node directly by calling `React.findDOMNode(this.refs.myInput)`.
 
 
-## The ref Callback Attribute
-
-The `ref` attribute can be a callback function instead of a name.  This callback will be executed immediately after the component is mounted.  The referenced component will be passed in as a parameter, and the callback function may use the component immediately, or save the reference for future use (or both).
-
-It's as simple as adding a `ref` attribute to anything returned from `render` by using an ES6 arrow function:
+## A Complete Example
+In order to get a reference to a React component, you can either use `this` to get the current React component, or you can use refs to refer to a component you own. They work like this:
 
-```html
-  render: function() {
-    return <TextInput ref={(c) => this._input = c} />;
-  },
-  componentDidMount: function() {
-    this._input.focus();
+```javascript
+var MyComponent = React.createClass({
+  handleClick: function() {
+    // Explicitly focus the text input using the raw DOM API.
+    this.myTextInput.focus();
   },
-```
-
-or
-
-```html
   render: function() {
+    // The ref attribute adds a reference to the component to
+    // this.refs when the component is mounted.
     return (
-      <TextInput
-        ref={function(input) {
-          if (input != null) {
-            input.focus();
-          }
-        }} />
+      <div>
+        <input type="text" ref="(ref) => this.myTextInput = ref" />
```

Comment:
Actually this should just be `ref="myTextInput"` since we're talking about string refs.